core/src/main/scala/caliban/validation/DefaultValue.scala

```diff
@@ -0,0 +1,131 @@
+package caliban.validation
+
+import caliban.CalibanError.ValidationError
+import caliban.InputValue.VariableValue
+import caliban.Value.NullValue
+import caliban.execution.{ ExecutionRequest, Field => F }
+import caliban.introspection.Introspector
+import caliban.introspection.adt._
+import caliban.introspection.adt.__TypeKind._
+import caliban.parsing.SourceMapper
+import caliban.parsing.adt.Definition.ExecutableDefinition.{ FragmentDefinition, OperationDefinition }
+import caliban.parsing.adt.Definition.{ TypeSystemDefinition, TypeSystemExtension }
+import caliban.parsing.adt.OperationType._
+import caliban.parsing.adt.Selection.{ Field, FragmentSpread, InlineFragment }
+import caliban.parsing.adt.Type.NamedType
+import caliban.parsing.adt._
+import caliban.schema.{ RootSchema, RootSchemaBuilder, RootType, Types }
+import caliban.{ InputValue, Rendering, Value }
+import caliban.parsing.Parser
+import zio.IO
+import caliban.schema.ArgBuilder
+import caliban.InputValue.ListValue
+import caliban.Value.EnumValue
+import caliban.Value.StringValue
+import caliban.Value.FloatValue.BigDecimalNumber
+import caliban.Value.IntValue.BigIntNumber
+import caliban.Value.BooleanValue
+import caliban.Value.IntValue.IntNumber
+import caliban.Value.IntValue.LongNumber
+import caliban.Value.FloatValue.DoubleNumber
+import caliban.Value.FloatValue.FloatNumber
+import caliban.InputValue.ObjectValue
+
+object DefaultValue {
+  def validateDefaultValue(field: __InputValue): IO[ValidationError, Unit] =
+    field.defaultValue.map { v =>
+      for {
+        value <-
+          IO.fromEither(Parser.parseInputValue(v))
+            .mapError(e =>
+              ValidationError(
+                s"Failed to parse default value for input '${field.name}': ${e.msg}",
+                "The default value for a field must be written using GraphQL input syntax."
+              )
+            )
+        _     <- Validator.validateInputValues(field, value)
```

Comment:
Not loving the circular dependency here, but maybe splitting the Validation up more is for another time.

Reply:
Can it make an infinite loop?

Reply:
No. I just have a very strong dislike towards circular dependecies. Might be Golang PTSD. 😅